test/mocks/MockImplementation.sol

```diff
@@ -121,3 +121,22 @@ contract RevertingInitializerMockImplementation is MockImplementation {
         revert("InitializerReverted");
     }
 }
+
+/**
+ * @dev Mock implementation that returns ERC1271_MAGIC_VALUE with extra data
+ */
+contract MockImplementationWithExtraData is MockImplementation {
+    function isValidSignature(
```

Comment:
nit: could we construct with an arg to make the extra data fuzzable (replacing hardcoded "deadbeef")

Reply:
doing this causes issues with the `isValidSignature` override which is pure but can't be marked as pure when accessing storage variables nor can it take an additional arg